src/walletConnect/screens/WalletConnectRequest.tsx

```diff
@@ -0,0 +1,82 @@
+import { StackScreenProps } from '@react-navigation/stack'
+import React, { useEffect } from 'react'
+import { useTranslation } from 'react-i18next'
+import { ActivityIndicator, StyleSheet, Text, View } from 'react-native'
+import { WalletConnectPairingOrigin } from 'src/analytics/types'
+import { headerWithBackButton, noHeader } from 'src/navigator/Headers'
+import { Screens } from 'src/navigator/Screens'
+import { StackParamList } from 'src/navigator/types'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import ActionRequest from 'src/walletConnect/screens/ActionRequest'
+import ConnectionTimedOut from 'src/walletConnect/screens/ConnectionTimedOut'
+import SessionRequest from 'src/walletConnect/screens/SessionRequest'
+
+type Props = StackScreenProps<StackParamList, Screens.WalletConnectRequest>
+
+enum ContentType {
+  Loading,
+  ConnectionRequest,
+  ActionRequest,
+  TimeOut,
+}
+
+function WalletConnectRequest({ navigation, route }: Props) {
+  const { t } = useTranslation()
+  const { pendingAction, timedOut, pendingSession, origin } = route.params
+  const fromScan = origin === WalletConnectPairingOrigin.Scan
+
+  let displayContent: ContentType = ContentType.Loading
+  if (timedOut) {
+    displayContent = ContentType.TimeOut
+  } else if (pendingSession) {
+    displayContent = ContentType.ConnectionRequest
+  } else if (pendingAction) {
+    displayContent = ContentType.ActionRequest
+  }
+
+  useEffect(() => {
+    const isLoading = !timedOut && !pendingSession && !pendingAction
+    navigation.setOptions(isLoading ? headerWithBackButton : noHeader)
```

Comment:
this is hacky, needed because only the loading screen has a back button right now (the e2e tests require this). this hack will be removed in the next iteration because the bottom sheet will provide a way of exiting the loading state